tests/helpers/fixtures.py

```diff
@@ -682,12 +673,13 @@ def download_stub(win_registry, tmpdir, stubs):
 
 
 @pytest.fixture
-def web_history(fake_save_manager, tmpdir, init_sql, config_stub, stubs,
+def web_history(fake_save_manager, tmpdir, data_tmpdir, config_stub, stubs,
                 monkeypatch):
     """Create a WebHistory object."""
     config_stub.val.completion.timestamp_format = '%Y-%m-%d'
     config_stub.val.completion.web_history.max_items = -1
-    web_history = history.WebHistory(stubs.FakeHistoryProgress())
+    db = sql.Database(str(data_tmpdir / 'history.db'))
```

Comment:
do we need to `yield` and call `db.close()` after the test ends?

Reply:
I didn't know exactly what this did so I took a look at pytest's documentation. I think this is not strictly needed, as we don't hold references to the `QSqlDatabase` objects and they are cleaned up automatically by Qt.

Reply:
But instantiating an `sql.Database` calls `QSqlDatabase.addDatabase` which stores them in what's essentially global state - so surely we should take care of cleaning that up properly after each test, no?

Reply:
Oh, yeah, now I see what you mean. I was using unique filenames and letting Qt clean all of them up after the tests finished. Having the fixture do the cleaning would make more sense, I'll adjust it.